src/app/relogin-popup/relogin-popup.service.ts

```diff
@@ -0,0 +1,56 @@
+import { Injectable, ErrorHandler } from '@angular/core';
+import { HttpClient, HttpHeaders, HttpParams, HttpErrorResponse } from '@angular/common/http';
+import { Observable } from 'rxjs';
+
+import { Response } from '../../../api-objects/GenericResponse';
+import { catchError, map, tap } from 'rxjs/operators';
+import { Utilities } from '../common/utlilities';
+import { LoginSession } from '../../../api-objects/LoginSession';
+import { authorityTypes } from '../../../api-objects/UsersObject';
+import baseUrl from '../common/base-url';
+
+@Injectable()
+export class ReloginPopupService {
+
+    constructor(private http: HttpClient) { }
+
+    public login(username: string, password: string): Observable<null> {
+        return this.post<LoginSession>('login', { 'username': username, 'password': password })
+            .pipe(
+                tap((response) => {
+
+                    Utilities.setSessionStorage('userId', response.data.user.toString());
+                    Utilities.setSessionStorage('sessionId', response.data.session.toString());
+                    console.log(authorityTypes[response.data.authorityType]);
+                    Utilities.setSessionStorage('permissions', authorityTypes[response.data.authorityType]);
+                }),
+                map<Response<LoginSession>, null>(_ => null)
+            );
+    }
+
+    private post<T>(urlExtension: string, body: any, queryParams?: Map<string, string>): Observable<Response<T>> {
+        return this.http.post<Response<T>>(baseUrl + urlExtension, body, this.createHeaders(queryParams));
```

Comment:
Everything seems okay to me. The only thing I'll suggest is a comment or something better explain what this is for. Even in the context of ReloginPopupService, it's not clear to me what the need is for this post. Still though, not something that would prevent a merge in my opinion.